src/lib/tecs/TECS.cpp

```diff
@@ -41,241 +46,403 @@ using math::constrain;
 using math::max;
 using math::min;
 
-static constexpr float DT_MIN = 0.001f;	///< minimum allowed value of _dt (sec)
-static constexpr float DT_MAX = 1.0f;	///< max value of _dt allowed before a filter state reset is performed (sec)
+// TODO there seems to be an inconsistent definition of IAS/CAS/EAS/TAS
+// TODO Recheck the timing.
+void TECSAirspeedFilter::initialize(const float equivalent_airspeed)
+{
 
-/**
- * @file TECS.cpp
- *
- * @author Paul Riseborough
- */
 
-/*
- * This function implements a complementary filter to estimate the climb rate when
- * inertial nav data is not available. It also calculates a true airspeed derivative
- * which is used by the airspeed complimentary filter.
- */
-void TECS::update_vehicle_state_estimates(float equivalent_airspeed, const float speed_deriv_forward,
-		bool altitude_lock, float altitude, float vz)
+	_airspeed_state.speed = equivalent_airspeed;
+	_airspeed_state.speed_rate = 0.0f;
+	_airspeed_rate_filter.reset(0.0f);
+}
+
+void TECSAirspeedFilter::update(const float dt, const Input &input, const Param &param,
+				const bool airspeed_sensor_available)
 {
-	// calculate the time lapsed since the last update
-	uint64_t now = hrt_absolute_time();
-	float dt = fmaxf((now - _state_update_timestamp) * 1e-6f, DT_MIN);
+	// Input checking
+	if (!(PX4_ISFINITE(dt) && dt > FLT_EPSILON)) {
```

Comment:
```suggestion
	if (!(PX4_ISFINITE(dt) && dt < FLT_EPSILON)) {
```

Reply:
I think you missed that i negate the whole expression instead of only the ISFINITE part?